test/e2e_node/hugepages_test.go

```diff
@@ -90,145 +116,309 @@ func makePodToVerifyHugePages(baseName string, hugePagesLimit resource.Quantity)
 }
 
-func runHugePagesTests(f *framework.Framework) {
-	ginkgo.It("should assign hugepages as expected based on the Pod spec", func() {
-		ginkgo.By("by running a G pod that requests hugepages")
-		pod := f.PodClient().Create(&v1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "pod" + string(uuid.NewUUID()),
-				Namespace: f.Namespace.Name,
-			},
-			Spec: v1.PodSpec{
-				Containers: []v1.Container{
-					{
-						Image: imageutils.GetPauseImageName(),
-						Name:  "container" + string(uuid.NewUUID()),
-						Resources: v1.ResourceRequirements{
-							Limits: v1.ResourceList{
-								v1.ResourceName("cpu"):           resource.MustParse("10m"),
-								v1.ResourceName("memory"):        resource.MustParse("100Mi"),
-								v1.ResourceName("hugepages-2Mi"): resource.MustParse("6Mi"),
-							},
-						},
+func getHugepagesTestPod(f *framework.Framework, limits v1.ResourceList, mounts []v1.VolumeMount, volumes []v1.Volume) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "hugepages-",
+			Namespace:    f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "container" + string(uuid.NewUUID()),
+					Image: busyboxImage,
+					Resources: v1.ResourceRequirements{
+						Limits: limits,
 					},
+					Command:      []string{"sleep", "3600"},
+					VolumeMounts: mounts,
 				},
 			},
-		})
-		podUID := string(pod.UID)
-		ginkgo.By("checking if the expected hugetlb settings were applied")
-		verifyPod := makePodToVerifyHugePages("pod"+podUID, resource.MustParse("6Mi"))
-		f.PodClient().Create(verifyPod)
-		err := e2epod.WaitForPodSuccessInNamespace(f.ClientSet, verifyPod.Name, f.Namespace.Name)
-		framework.ExpectNoError(err)
-	})
+			Volumes: volumes,
+		},
+	}
 }
 
 // Serial because the test updates kubelet configuration.
-var _ = SIGDescribe("HugePages [Serial] [Feature:HugePages][NodeFeature:HugePages]", func() {
+var _ = SIGDescribe("HugePages [Serial] [Feature:HugePages][NodeSpecialFeature:HugePages]", func() {
+	var (
+		testpod   *v1.Pod
+		limits    v1.ResourceList
+		mounts    []v1.VolumeMount
+		volumes   []v1.Volume
+		hugepages map[string]int
+	)
+
 	f := framework.NewDefaultFramework("hugepages-test")
 
-	ginkgo.Context("With config updated with hugepages feature enabled", func() {
-		ginkgo.BeforeEach(func() {
-			ginkgo.By("verifying hugepages are supported")
-			if !isHugePageSupported() {
-				e2eskipper.Skipf("skipping test because hugepages are not supported")
+	setHugepages := func() {
+		for hugepagesResource, count := range hugepages {
+			size := resourceToSize[hugepagesResource]
+			ginkgo.By("Verifying hugepages are supported")
+			if !isHugePageSupported(size) {
+				e2eskipper.Skipf("skipping test because hugepages of size %d not supported", size)
 				return
 			}
-			ginkgo.By("configuring the host to reserve a number of pre-allocated hugepages")
+
+			ginkgo.By("Configuring the host to reserve a number of pre-allocated hugepages")
```

Comment:
Same here if we can include size and count

Reply:
done